[leongxy] iP

leongxingyu opened the pull request.

leongxingyu committed:
Rename chatbot
Implement Greet and Exit

leongxingyu committed:
Rename chatbot, implement Greet and Exit

leongxingyu committed:
Rename chatbot, implement Greet and Exit

leongxingyu committed:
Rename chatbot, implement Greet and Exit

leongxingyu committed:
Implement Echo

leongxingyu committed:
Implement store tasks and display when requested

leongxingyu committed:
Add Task class for each class

leongxingyu committed:
Add ability to mark task as done

leongxingyu committed:
Add ability to unmark tasks

leongxingyu committed:
Edit switch statements indentation

leongxingyu committed:
Add Task, Todo, Deadline, Event class

leongxingyu committed:
Unify spacing in toString() methods

Edit spacing for Deadline, Event and Task classes such that it follows the "[ ] [ ] taskname" format.

leongxingyu committed:
Modify code to use Todo, Deadline and Event classes

Add 3 new cases, "todo", "deadline" and "event" to make use of the respective classes created.

leongxingyu committed:
Update code to use Todo, Deadline, Event class

leongxingyu committed:
Improve code quality

leongxingyu committed:
Change code to comply with coding standard

leongxingyu committed:
Add TextUiTesting

Daviancold commented on src/main/java/Gab.java:
```diff
@@ -0,0 +1,63 @@
+
+    public static void checkKeywords (String task) { //going to input task description from getTask here
+        String[] taskAction = task.split(" ");
+        String action = taskAction[0];
```
It would be good if you added some String Cleaning such as changing all to lower cases and trimming to make the input processing more robust.

Daviancold commented on src/main/java/Gab.java:
```diff
@@ -0,0 +1,63 @@
+
+public class Gab {
+
+    private static Task[] taskList = new Task[100]; //array of Task instances
```
Can consider changing 100 to a constant variable

Daviancold commented on src/main/java/Gab.java:
```diff
@@ -0,0 +1,63 @@
+            getTask();
+        } else {
+            switch (action) {
+            case "bye":
```
Great work ensuring correct indentation for case statements!

Daviancold commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,34 @@
+        isDone = true;
+    }
+
+    public void markAsNotDone() { isDone = false; }
```
Try to avoid single line functions

json-bot-boop replied:
Maybe can combine these 2 methods into 1 single method

Daviancold commented on src/main/java/Gab.java:
```diff
@@ -0,0 +1,63 @@
+        Scanner in = new Scanner(System.in);
+        System.out.println("\tWhat do you want to do?: ");
+        taskDescription = in.nextLine();
+        checkKeywords(taskDescription); //check if its un mark/mark/exit/list
```
Comments here are appropriately placed

Daviancold commented on src/main/java/Ui.java:
```diff
@@ -0,0 +1,33 @@
+    public static void listTask(Task[] taskList) { //using the array to list the tasks
+        System.out.println("All your tasks are here");
+        for (int i = 0; i < Task.getTaskCount(); i++) {
+            System.out.println((i+1) + "." + taskList[i].toString());
```
Addition should be delimited by blank spaces e.g. (i + 1)

Daviancold reviewed (commented):
Clean code generally, follows proper convention.

ShyamKrishna33 commented on src/main/java/CommandHandler.java:
```diff
@@ -0,0 +1,50 @@
+
+    public static void setDeadline (String task, Task[] taskList) {
+        int deadlineIndex = task.indexOf("/by"); //gives index of /by
+        String deadline = task.substring(deadlineIndex + 4);
```
Avoid using magic numbers. Alternatively, you can make it a constant or add a comment.

ShyamKrishna33 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,14 @@
+
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + " " + "(by: " + by + ")"; //superclass is task
```
I would recommend to have the label 'D' as a static constant outside the method.

ShyamKrishna33 commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,23 @@
+
+    @Override
+    public String toString() {
+        return "[E]" + super.toString() + " " + "(from: " + startTime + " to:" + endTime + ")";
```
I would recommend to have the label 'E' as a static constant outside the method.

ShyamKrishna33 commented on src/main/java/ToDo.java:
```diff
@@ -0,0 +1,11 @@
+
+    @Override
+    public String toString() {
+        return "[T]" + super.toString();
```
I would recommend to have the label 'T' as a static constant outside the method.

ShyamKrishna33 reviewed (commented):
Overall, the code quality is maintained good.

leongxingyu committed:
Add Command packages

leongxingyu committed:
Add Exception packages

leongxingyu committed:
Add static constants based on PR

leongxingyu committed:
Revert "Add static constants based on PR"

This reverts commit 78e1dfee04bd7d4e55bde8d41475ac44edc4f792.

leongxingyu committed:
Revert "Add Exception packages"

This reverts commit ffff7c9946e59901ac03390377ae1e47ae489899.

leongxingyu committed:
Revert "Add Command packages"

This reverts commit aa00d0a8c5632b263bfc7c3930896abbfaf747da.

leongxingyu committed:
Revert "Add Exception packages"

This reverts commit ffff7c9946e59901ac03390377ae1e47ae489899.

leongxingyu committed:
Revert "Add Command packages"

This reverts commit aa00d0a8c5632b263bfc7c3930896abbfaf747da.

leongxingyu committed:
Add Command packages

leongxingyu committed:
Add exception packages

leongxingyu committed:
Merge branch 'branch-Level-5'

json-bot-boop commented on src/main/java/Ui.java:
```diff
@@ -0,0 +1,33 @@
+        System.out.println("_______________________");
+        System.out.println("Weehoo! Enjoy the event!");
+        System.out.println(eventTask);
+    }
```
Maybe can combine these 3 methods into 1 method

json-bot-boop commented on src/main/java/CommandHandler.java:
```diff
@@ -0,0 +1,50 @@
+    }
+
+    public static void setDeadline (String task, Task[] taskList) {
+        int deadlineIndex = task.indexOf("/by"); //gives index of /by
```
Try not to repeat in comments information that is already explained in the code. This code quality will be taught in future lecture.

json-bot-boop commented on src/main/java/CommandHandler.java:
```diff
@@ -0,0 +1,50 @@
+public class CommandHandler {
+
+    public static void setToDo(String task, Task[] taskList) {
+        String todoName = task.substring(5);
```
Try to avoid magic number

json-bot-boop commented on src/main/java/Gab.java:
```diff
@@ -0,0 +1,63 @@
+        String action = taskAction[0];
+
+        if (action.equals("mark") || action.equals("unmark")) {
+            CommandHandler.markTask(task, taskList); //need display as [ ][X] task name
```
Remove developer comments from the code

json-bot-boop commented on src/main/java/Gab.java:
```diff
@@ -0,0 +1,63 @@
+                break;
+            case "event":
+                CommandHandler.setEvent(task, taskList);
+                break;
```
Can consider the case if there are no arguments present when entering the command

json-bot-boop replied:
Try to add include the default branch for the switch. This ensures that all possible outcomes have been considered at the branching point. This code quality will be taught in future lectures.

json-bot-boop commented on src/main/java/Gab.java:
```diff
@@ -0,0 +1,63 @@
+                        + "|  |  __   ____  |  |__\n"
+                        + "|  | |_  |/  _   |  -   \\ \n"
+                        + "|  |__|  |  (_|  | |_)   |\n"
+                        + "\\_______ |__ ,_ |_.___ /\n";
```
Maybe can consider putting this variable as a constant variable

json-bot-boop commented on src/main/java/Gab.java:
```diff
@@ -0,0 +1,63 @@
+                CommandHandler.setEvent(task, taskList);
+                break;
+            }
+            getTask();
```
Can considering calling the getTask method outside the if-else statement

json-bot-boop reviewed (commented):
Overall, it is good but have some code quality and coding standard issue.

leongxingyu committed:
Edit code quality

leongxingyu committed:
Merge branch 'branch-Level-5'

leongxingyu committed:
Remove command and exception packages

leongxingyu committed:
Merge branch 'branch-Level-5'

leongxingyu committed:
Add Command and Exception classes

leongxingyu committed:
Merge branch 'branch-Level-5'

leongxingyu committed:
Edit files to use command classes

leongxingyu committed:
Change logo into constant variable

leongxingyu committed:
Implement ArrayList

leongxingyu committed:
Modify classes to use ArrayList

leongxingyu committed:
Modify code to comply with coding standard

leongxingyu committed:
Add Storage function

leongxingyu committed:
Add Delete Command

leongxingyu committed:
Merge branch 'branch-Level-6'

leongxingyu committed:
Resolve merge conflicts from branch-Level-7

# Conflicts:
#	src/main/java/Parser.java

leongxingyu committed:
Remove spaces and unused methods

leongxingyu committed:
Create JAR file

leongxingyu committed:
Implement A-Packages

leongxingyu committed:
Merge branch 'branch-A-Packages'

leongxingyu committed:
Update code to implement jar file

leongxingyu committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/gab.txt

leongxingyu committed:
Implement jar file draft 2

leongxingyu committed:
Update Gab.java for storage file

leongxingyu committed:
Edit filepath for storage

leongxingyu committed:
Change interface to abstract class

leongxingyu committed:
Add Find Command and print found tasks

leongxingyu committed:
Edit parser to handle error spaces

Replace .split(" ") by .split("\\s+") so that code can handle cases with extra spaces between command and task name

leongxingyu committed:
Merge branch 'branch-Level-9'

leongxingyu committed:
Standardise Ui and create printLine() method

Standardise printLine() format for all Ui displays

leongxingyu committed:
Change command class parameters

Reduce command class parameters to just taskList and the String input was not used in other abstract classes

leongxingyu committed:
Add Java Doc

leongxingyu committed:
Merge pull request #1 from leongxingyu/branch-Level-A-JavaDoc

Add Java Doc

leongxingyu committed:
Update code to fix bugs

Modify positions of printLine() function to standardise between all classes
Fix bug where the deadline (by: ) does not dislay properly when retrieving from storage

leongxingyu committed:
Add Parser for Bye and List commands

leongxingyu committed:
Add User Guide

leongxingyu committed:
Merge pull request #2 from leongxingyu/branch-UserGuide

Add User Guide

leongxingyu committed:
Add JavaDoc for listTask and exitBot

leongxingyu committed:
Standardise spacing for Java Doc

leongxingyu committed:
Add hyperlinks in README

leongxingyu committed:
Merge pull request #3 from leongxingyu/branch-UserGuide

Add hyperlinks in README

leongxingyu committed:
Add Warning, FAQ and Command Summary in UG

leongxingyu committed:
Edit README spacings

leongxingyu committed:
Edit README for project template

leongxingyu committed:
Change Duke to Gab in project template
